modules/affile/directory-monitor-kqueue.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2017 Balabit
+ * Copyright (c) 2024 OneIdentity
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "directory-monitor-kqueue.h"
+#include "directory-monitor-content-comparator.h"
+#include "messages.h"
+
+#include <iv.h>
+#include <iv_event.h>
+#include <sys/event.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdio.h>
+
+typedef struct _DirectoryMonitorKQueue
+{
+  DirectoryMonitorContentComparator super;
+  struct iv_fd kqueue_fd;
+  gint dir_fd;
+} DirectoryMonitorKQueue;
+
+static void handle_kqueue_event(void *s)
+{
+  DirectoryMonitorKQueue *self = (DirectoryMonitorKQueue *)s;
+
+  struct kevent events[64];
+  struct timespec timeout = {0, 0}; // non-blocking
+
+  while (1)
+    {
+      int nevents = kevent(self->kqueue_fd.fd, NULL, 0, events,
+                           sizeof(events) / sizeof(events[0]), &timeout);
+      if (nevents < 0)
```

Comment:
shouldn't it be -1 as the docs explicitl says it is -1 on error?
https://man.freebsd.org/cgi/man.cgi?query=kevent&apropos=0&sektion=0&manpath=FreeBSD+9.0-RELEASE&arch=default&format=html

Reply:
or is it different on other BSD derivatives?

Reply:
I was this exact pattern on open() as well, so n/a

Reply:
i guess -1 is less than 0 ;)